server/main_server/src/main.cpp

```diff
@@ -0,0 +1,1342 @@
+            daemonize();
+        }
+
+        std::string pid_file = (irods::get_irods_runstate_directory() / "irods/irods-server.pid").string();
+        if (const auto iter = vm.find("pid-file"); std::end(vm) != iter) {
+            pid_file = std::move(iter->second.as<std::string>());
+        }
+
+        if (create_pid_file(pid_file) != 0) {
+            fmt::print(stderr, "Error: could not create PID file [{}].\n", pid_file);
+            return 1;
+        }
+    }
+    catch (const std::exception& e) {
+        fmt::print(stderr, "Error: {}\n", e.what());
+        return 1;
+    }
+
+    try {
+        if (!validate_configuration()) {
+            return 1;
+        }
+
+        const auto config_file_path = irods::get_irods_config_directory() / "server_config.json";
+        irods::server_properties::instance().init(config_file_path.c_str());
+        irods::environment_properties::instance(); // Load the local environment file.
+
+        // TODO Consider removing the need for these along with all options.
+        // All logging should be controlled via the new logging system.
+        rodsLogLevel(LOG_NOTICE);
+    	rodsLogSqlReq(0);
+
+        init_logger(getpid(), write_to_stdout, enable_test_mode);
+
+        // Setting up signal handlers here removes the need for reacting to shutdown signals
+        // such as SIGINT and SIGTERM during the startup sequence.
+        if (setup_signal_handlers() == -1) {
+            log_server::error("{}: Error setting up signal handlers for main server process.", __func__);
+            return 1;
+        }
+
+        if (wait_for_external_dependent_servers_to_start() != 0) {
+            if (g_terminate || g_terminate_graceful) {
+                return 0;
+            }
+
+            return 1;
+        }
+
+        if (const auto [up_to_date, db_vers] = check_catalog_schema_version(); !up_to_date) {
+            const auto msg = fmt::format("Catalog schema version mismatch: expected [{}], found [{}] in catalog", IRODS_CATALOG_SCHEMA_VERSION, db_vers);
+            log_server::critical(msg);
+            fmt::print(fmt::runtime(msg));
+            return 1;
+        }
+
+        log_server::info("{}: Initializing shared memory for main server process.", __func__);
+
+        namespace hnc = irods::experimental::net::hostname_cache;
+        hnc::init("irods_hostname_cache", irods::get_hostname_cache_shared_memory_size());
+        irods::at_scope_exit deinit_hostname_cache{[] { hnc::deinit(); }};
+
+        namespace dnsc = irods::experimental::net::dns_cache;
+        dnsc::init("irods_dns_cache", irods::get_dns_cache_shared_memory_size());
+        irods::at_scope_exit deinit_dns_cache{[] { dnsc::deinit(); }};
+
+        // TODO These directories should be created by the packaging.
+        fs::create_directories(irods::get_irods_stacktrace_directory().c_str());
+        fs::create_directories((irods::get_irods_home_directory() / "log/proc").c_str());
```

Comment:
Open an issue for this and handle it later.

Reply:
@SwooshyCueb What's your thoughts on the TODO comment?

This works, but it seems like the packaging should take care of that.

Reply:
Going to open an issue for this so it can be addressed in a separate PR.

This isn't a blocker for this PR.

Reply:
Created #8014

Reply:
Attached issue number to TODO.